### PR TITLE
Add staging commands

### PR DESCRIPTION
**Pull request by sampierson**
```
hca staging select <urn_or_alias>
hca staging upload <file>
hca staging areas
hca staging help
```

### CODE REVIEW

File: hca/staging/cli/select_command.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+from .config_store import ConfigStore
+from .staging_area_urn import StagingAreaURN
+
+
+class SelectCommand:
+
+    def __init__(self, args):
+        self.config = ConfigStore()
+        if re.search(':', args.urn_or_alias):  # URN
```
Comment (ttung) [inline hca/staging/cli/select_command.py]: `if args.urn_or_alias.find(":") >= 0:`

File: hca/staging/cli/select_command.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    def _save_and_select_area_by_urn(self, urn_string):
+        urn = StagingAreaURN(urn_string)
+        self.config.add_area(urn)
+        self._select_area(urn.uuid)
+
+    def _select_area_by_alias(self, alias):
+        matching_areas = [uuid for uuid in self.config.areas() if re.match(alias, uuid)]
+        if len(matching_areas) == 0:
+            print("Sorry I don't recognize area \"%s\"" % (alias,))
```
Comment (ttung) [inline hca/staging/cli/select_command.py]: If you're going to do the `(xxx,)` thing for a single element tuple, you should be consistent about doing it everywhere.

File: hca/staging/cli/select_command.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        elif len(matching_areas) == 1:
+            self._select_area(matching_areas[0])
+        else:
+            print("\"%s\" matches more than one area, please provide more characters." % (alias,))
+
+    def _select_area(self, area_uuid):
+        self.config.select_area(area_uuid)
+        print("Staging area %s selected." % area_uuid)
+        print("You may refer to this staging area in future using the first few characters, e.g. \"%s\"" %
```
Comment (ttung) [inline hca/staging/cli/select_command.py]: s/in future/in the future/

File: hca/staging/cli/select_command.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+              area_uuid[0:3])
```
Comment (ttung) [inline hca/staging/cli/select_command.py]: strictly speaking, this is not true.  you'd need to run it through `_select_area_by_alias` to see if it's a unique prefix.

File: hca/staging/cli/staging_area_urn.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import json
+import base64
```
Comment (ttung) [inline hca/staging/cli/staging_area_urn.py]: alphabetical.

**Comment by sampierson** [comment]
Addressed all of Tony's comments:
- alphabetize imports
- use `str.find()` instead of `re.search`
- use % tuples consistently
- rewording
- return truly unique aliases

**Pull request merged**